### PR TITLE
Hotfix/auth fix

### PR DESCRIPTION
**Pull request by gillescoolen**

### CODE REVIEW

File: src/common/decorators/public.decorator.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import { SetMetadata } from '@nestjs/common';
+
+export const Public = () => SetMetadata('isPublic', true);
```
Comment (rauldeheer) [inline src/common/decorators/public.decorator.ts]: Add constant

File: src/auth/guards/jwt-auth.guard.ts [hunk 2]
```diff
@@ -1,5 +1,18 @@
...
 
 @Injectable()
-export class JwtGuard extends AuthGuard('jwt') {}
+export class JwtGuard extends AuthGuard('jwt') {
+  constructor(private reflector: Reflector) {
+    super();
+  }
+
+  canActivate(context: ExecutionContext): boolean {
+    const isPublic = this.reflector.get<boolean>('isPublic', context.getHandler());
```
Comment (rauldeheer) [inline src/auth/guards/jwt-auth.guard.ts:13]: Use constant

File: src/auth/guards/jwt-auth.guard.ts [hunk 3]
```diff
@@ -1,5 +1,18 @@
...
+
+    if (isPublic) return true;
+
+    super.canActivate(context);
```
Comment (rauldeheer) [inline src/auth/guards/jwt-auth.guard.ts]: Add return

File: src/common/interceptors/xml.interceptor.ts [hunk 4]
```diff
@@ -9,7 +9,8 @@ export class XMLInterceptor<T> implements NestInterceptor<T, string> {
     const regex = /\/xml/gi;
     const ctx = context.switchToHttp();
     const request = ctx.getRequest<Request>();
-    const requestedXML = request.headers['content-type'] !== undefined && request.headers['content-type'].match(regex);
+    const header = request.headers['content-type'];
```
Comment (rauldeheer) [inline src/common/interceptors/xml.interceptor.ts]: Use the `accept` header instead

**Review by rauldeheer — CHANGES_REQUESTED** [review]

File: src/common/decorators/public.decorator.ts (hunk 1 above)
Comment (gillescoolen) [reply src/common/decorators/public.decorator.ts]: ?????????????????

File: src/common/interceptors/xml.interceptor.ts (hunk 4 above)
Comment (gillescoolen) [reply src/common/interceptors/xml.interceptor.ts]: Fixed

File: src/common/decorators/public.decorator.ts (hunk 1 above)
Comment (rauldeheer) [reply src/common/decorators/public.decorator.ts]: According to the NestJS documentation you should export the string `isPublic` to a constant for usage in other classes.

Comment (gillescoolen) [reply src/common/decorators/public.decorator.ts]: Okeeeeeeeeeeee gedaan

**Review by rauldeheer — APPROVED** [review]
Ok boomer

**Pull request merged**